examples/large_models/ipex_llm_int8/llm_handler.py

```diff
@@ -0,0 +1,619 @@
+            (4) throws error if both 1 and 2 are enabled 
+            """
+            if self.ipex_smooth_quantization and self.ipex_weight_only_quantization:
+                logger.error("Can't enable both SQ and WoQ, enable only one of them")
+                exit(1)
+        
+            # Clear the cache dir if needed 
+            if self.clear_cache_dir and os.path.exists(self.quantized_model_path):
+                os.remove(self.quantized_model_path)
+
+            if os.path.exists(self.quantized_model_path):
+               # this skips all the optimizations and goes to end where we load the model 
+               logger.info("A previously quantized model is loaded, if you want to re-quantize the model, enable clear_cache_dir on model config file")
+
+            # lets implement the WOQ 
+            elif self.ipex_weight_only_quantization:
+                weight_dtype = torch.quint4x2 if self.woq_dtype == "INT4" else torch.qint8
+
+                if self.lowp_mode == "INT8":
+                    lowp_mode = ipex.quantization.WoqLowpMode.INT8
+                elif self.lowp_mode == "FP32":
+                    lowp_mode = ipex.quantization.WoqLowpMode.NONE
+                elif self.lowp_mode == "FP16":
+                    lowp_mode = ipex.quantization.WoqLowpMode.FP16
+                elif self.lowp_mode == "BF16":
+                    lowp_mode = ipex.quantization.WoqLowpMode.BF16
+                else:
+                    lowp_mode = ipex.quantization.WoqLowpMode.BF16
+
+                act_quant_mode_dict = {
+                    "PER_TENSOR": ipex.quantization.WoqActQuantMode.PER_TENSOR,
+                    "PER_IC_BLOCK": ipex.quantization.WoqActQuantMode.PER_IC_BLOCK,
+                    "PER_BATCH": ipex.quantization.WoqActQuantMode.PER_BATCH,
+                    "PER_BATCH_IC_BLOCK": ipex.quantization.WoqActQuantMode.PER_BATCH_IC_BLOCK,
+                }
+
+                qconfig = ipex.quantization.get_weight_only_quant_qconfig_mapping(
+                    weight_dtype=weight_dtype,
+                    lowp_mode=lowp_mode,
+                    act_quant_mode=act_quant_mode_dict[self.act_quant_mode],
+                    group_size=self.group_size,
+                )
+                
+                # low precision checkpoint can be loaded, but we're considering there isn't any 
+                low_precision_checkpoint = None
+                self.user_model = ipex.llm.optimize(
+                    self.user_model.eval(),
+                    dtype=self.amp_dtype,
+                    quantization_config=qconfig,
+                    inplace=True,
+                    low_precision_checkpoint=low_precision_checkpoint,
+                    deployment_mode=False,
+                )
+                logger.info("The model conversion completed, now tracing the quantized model")
+
+                example_inputs = self.get_example_inputs()
+
+                with torch.no_grad(), torch.cpu.amp.autocast(
```

Comment:
nit: The following lines (tracing and saving the model) are equal to all 3 conditions and could be replace by a single appearance after the if-else

Reply:
replaced by single `trace_and_export` function